datafusion/sqllogictest/test_files/struct.slt

```diff
@@ -63,10 +69,10 @@ query TT
 explain select struct(a, b, c) from values;
 ----
 logical_plan
-Projection: struct(values.a, values.b, values.c)
+Projection: named_struct(Utf8("c0"), values.a, Utf8("c1"), values.b, Utf8("c2"), values.c)
 --TableScan: values projection=[a, b, c]
 physical_plan
-ProjectionExec: expr=[struct(a@0, b@1, c@2) as struct(values.a,values.b,values.c)]
+ProjectionExec: expr=[named_struct(c0, a@0, c1, b@1, c2, c@2) as named_struct(Utf8("c0"),values.a,Utf8("c1"),values.b,Utf8("c2"),values.c)]
```

Comment:
Yeah, I also have the same concern as @alamb said in https://github.com/apache/arrow-datafusion/pull/9743#discussion_r1536800330. It seems that from here, the current `struct` UDF will never be called because calling `struct` will rewrite into `named_struct` UDF 👀

Reply:
I agree

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/9839 for follow up